### PR TITLE
Save the planet by not reading images for size

### PR DESCRIPTION
**Pull request by botcs**
```
In [7]: %timeit Image.open("datasets/coco/train2017/000000000009.jpg").size                           
34.9 µs ± 59.6 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [8]: %timeit imagesize.get("datasets/coco/train2017/000000000009.jpg")                             
9.43 µs ± 64.2 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

https://images.app.goo.gl/Msptv7xcCVsXJ5zh8

### CODE REVIEW

File: detectron2/data/datasets/coco.py [hunk 1]
```diff
@@ -262,12 +263,10 @@ def file2id(folder_path, file_path):
 
     dataset_dicts = []
     for (img_path, gt_path) in zip(input_files, gt_files):
+        w, h = imagesize.get(gt_path)
```
Comment (bendidi) [inline detectron2/data/datasets/coco.py]: with this we probably lose the flexibility of `PathManager` ?
For me for example I've registred multiple new handlers for PathManager and my images are all online.

maybe add `local_path = PathManager. get_local_path(gt_path)` before ?

Comment (botcs) [reply detectron2/data/datasets/coco.py]: well that sounds good :)

Comment (botcs) [reply detectron2/data/datasets/coco.py]: @bendidi Updated in a31f70b

**Comment by ppwwyyxx** [comment]
Could you add `imagesize` as a dependency in `setup.py`.
btw in the future with some dataloader changes I hope to remove the need to include width/height in the dicts.

**Comment by botcs** [comment]
`imagesize` was already installed somehow, probably it was a dependency for a pip package.
Do you want me to still add it to `setup.py` explicitly?

**Comment by ppwwyyxx** [comment]
Perhaps it was already a dependency, but it's still good to declare it explicitly.

**Comment by botcs** [comment]
Done in 84ac5f6

**Pull request closed without merging**